### ISSUE TITLE
PHP Doctrine with SQL Server Always On

### ISSUE THREAD

**Issue by fabionvs** [doctrine/doctrine2#6489]
Hello,
I am developing an application using Symfony 3 (php 7.1) with SQL Server database with AlwaysOn. AlwaysOn is a SQL server support for high-availability and disaster recovery, and Doctrine 2 is not working well with this paradigm.

I've developed a CRUD to test it for the infrastructure section at my job that implemented this database. At the first time, it was working correctly and persisting ok. Although, when they change the database for other one, like when happens if the first one stops working, automatically go to the other database with the same structure. When it happens I receive an error message from doctrine:

    

> Fatal error: Invalid handle returned. in C:\xampp\htdocs\testesql\vendor\doctrine\dbal\lib\Doctrine\DBAL\Driver\PDOConnection.php on line 43

It looks like Doctrine caches something that is not working with this paradigm of database. I am using Microsoft SQL Server driver (pdo_sqlsrv). Could someone help me? I would be glad.

**Comment by Ocramius** [comment]
@fabionvs any way to reproduce this? What are you passing to the connection?

 >  when they change the database for other one, like when happens if the first one stops working, automatically go to the other database with the same structure. When it happens I receive an error message from doctrine:

I can't understand the wording in this bit: can you rephrase it?

**Comment by fabionvs** [comment]
Hello @Ocramius, thank you for your help!
The SQL Server database was implemented in a cluster. And, when the main database stops working, or if they change the instance, it goes to another backup instance while using the same connection. And, for testing propourses they changed the instance to see if the application still works, but is not working after they changed the instance.

**Comment by Ocramius** [comment]
That's most likely something that needs fixing within `pdo_sqlsrv`, not in doctrine, sorry :-\

Closing as `can't fix` here.

**Comment by Ocramius** [comment]
Note: removed the `Bug` label because this is a `pdo_sqlsrv` issue that isn't a bug in doctrine itself.
